pkg/image/registry.go

```diff
@@ -0,0 +1,33 @@
+package image
+
+import (
+	"context"
+	"io"
+)
+
+// At the high-level, we need to be able to do the following w/ images
+// pull(ref)
+// push(ref)
+// unpack(ref, writer)
+// build(ref, reader) => new ref
+
+type Registry interface {
+	// Pull fetches and stores an image by reference.
+	Pull(ctx context.Context, ref string) error
```

Comment:
One thing I like is using custom types for specificity such a `type ImageReference string` and then in the interface it would take in a `ref ImageReference` type as the arg. I'm not sure how appropriate it is here but something worth noting as it makes the code more readable IMO.

Reply:
I wouldn't insist on it, but I also appreciate the extra compile-time safety of derived types for this kind of thing.

Reply:
Okay -- I'll change this